### PR TITLE
Add coverage rate test for predefined GUC

### PR DESCRIPTION
**Pull request by weinan003**
In gpdb, we force predefined guc name populate in either sync_guc_name.h
or unsync_guc_name.h. However, if a new guc introduce in and forget to add
to the lists. An error dose not raised until runtime. It is a little bit
late.

Using the unit test mechanism to do coverage rate test prevent this case happened.

## Here are some reminders before you submit the pull request
- [x] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [x] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: src/backend/utils/misc/test/guc_gp_test.c [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+			sync_guc_num,
+			sizeof(char *),
+			guc_array_compare);
+	if (!res)
+	{
+		char *res = (char *) bsearch((void *) &conf->name,
+				(void *) unsync_guc_names_array,
+				unsync_guc_num,
+				sizeof(char *),
+				guc_array_compare);
```
Comment (paul-guo-) [inline src/backend/utils/misc/test/guc_gp_test.c]: Better provide details about which guc is missing in sync or unsync array.

Comment (weinan003) [reply src/backend/utils/misc/test/guc_gp_test.c]: Gotcha

File: src/backend/utils/misc/test/guc_gp_test.c [hunk 2]
```diff
@@ -34,6 +34,10 @@ static void assert_guc(struct config_generic *conf)
 				unsync_guc_num,
 				sizeof(char *),
 				guc_array_compare);
+
+		if ( res == NULL)
+			printf("GUC: '%s' does not exist in both list.\n", conf->name);
```
Comment (asimrp) [inline src/backend/utils/misc/test/guc_gp_test.c]: Minor suggestion: rephrase to `... does not exist in either list`

File: src/backend/utils/misc/test/guc_test.c [hunk 3]
```diff
@@ -117,14 +121,17 @@ test_enum_guc_coverage(void **state)
...
+	for (int i = 0; i < sync_guc_num; i++)
 	{
 		char *res = (char *) bsearch((void *) &(sync_guc_names_array[i]),
 				(void *) unsync_guc_names_array,
 				unsync_guc_num,
 				sizeof(char *),
 				guc_array_compare);
 
+		if ( res != NULL)
+			printf("GUC: '%s' exist in both list.\n", sync_guc_names_array[i]);
```
Comment (asimrp) [inline src/backend/utils/misc/test/guc_test.c]: Minor suggestion: rephrase to `... exists in both lists`

File: src/include/utils/sync_guc_name.h [hunk 4]
```diff
@@ -107,3 +107,36 @@
...
+		"default_transaction_read_only",
+		"dynamic_library_path",
+		"extra_float_digits",
+		"gp_allow_rename_relation_without_lock",
+		"gp_appendonly_compaction",
+		"gp_appendonly_compaction_threshold",
+		"gp_appendonly_verify_block_checksums",
+		"gp_appendonly_verify_write_block",
+		"gp_autostats_on_change_threshold",
+		"gp_cached_segworkers_threshold",
```
Comment (zhangh43) [inline src/include/utils/sync_guc_name.h]: gp_cached_segworkers_threshold only used on master.
I'm not sure whether we should pass client_encoding
We'd better involve more discussion on which guc need to be synced

**Pull request merged**